packages/components/src/components/PasswordCreationField/PasswordCreationField.tsx

```diff
@@ -0,0 +1,315 @@
+    >,
+    FlowComponentProps<HTMLInputElement> {
+  value?: string;
+  defaultValue?: string;
+  validationPolicy?: Policy;
+}
+
+export interface ResolvedPolicyValidationResult extends PolicyValidationResult {
+  isEmptyValueValidation: boolean;
+  isValid: boolean;
+  ruleResults: RuleValidationResult[];
+}
+
+/**
+ * @flr-generate all
+ * @flr-clear-props-context
+ */
+export const PasswordCreationField = flowComponent(
+  "PasswordCreationField",
+  (props) => {
+    const {
+      children,
+      value: valueFromProps,
+      defaultValue,
+      className,
+      ref,
+      isDisabled,
+      onChange,
+      isInvalid: invalidFromProps,
+      validationPolicy = defaultPasswordCreationPolicy,
+      isRequired,
+      ...rest
+    } = props;
+
+    const promiseQueue = usePromiseQueue();
+    const translate = useLocalizedStringFormatter(locales);
+    const generatePassword = useGeneratePassword(validationPolicy);
+
+    const [isPasswordRevealed, setIsPasswordRevealed] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const initialPolicyValidationState: ResolvedPolicyValidationResult = {
+      isEmptyValueValidation: true,
+      isValid: false,
+      complexity: {
+        min: validationPolicy.minComplexity,
+        actual: 4 as ComplexityScore,
+        warning: null,
+      },
+      ruleResults: [],
+    };
+
+    const [policyValidationResult, setPolicyValidationResult] = useState(
+      initialPolicyValidationState,
+    );
+
+    const stateFromValidationResult = getStateFromLatestPolicyValidationResult(
+      policyValidationResult,
+    );
+    let latestValidationErrorText = undefined;
+    if (stateFromValidationResult) {
+      const [translationKey, translationValues] = generateValidationTranslation(
+        stateFromValidationResult,
+      );
+      latestValidationErrorText = translate.format(
+        translationKey,
+        translationValues,
+      );
+    }
+
+    const isValidFromValidationResult =
+      !policyValidationResult.isEmptyValueValidation &&
+      stateFromValidationResult?.isValid;
+
+    const isInvalidFromValidationResult =
+      !policyValidationResult.isEmptyValueValidation &&
+      !stateFromValidationResult?.isValid;
+
+    const isInvalid = invalidFromProps || isInvalidFromValidationResult;
+
+    const valueControlType = useRef<"controlled" | "uncontrolled">(
+      valueFromProps === undefined ? "uncontrolled" : "controlled",
+    ).current;
+    const [uncontrolledValue, setUncontrolledValueRaw] = useState(
+      defaultValue ?? "",
+    );
+    const value =
+      valueControlType === "controlled" ? valueFromProps : uncontrolledValue;
+    const [bouncedValue, setDebouncedValue] = useDebounceValue(
+      value ?? "",
+      validationDebounceMilliseconds,
+    );
+    const setUncontrolledValue = (value: string) => {
+      setUncontrolledValueRaw(value);
+      setDebouncedValue(value);
+    };
+
+    useEffect(() => {
```

Comment:
Instead of the `PromiseQueue` you could detect aborted promises locally with the `useEffect` cleanup function.

This hook should do the same job:

```tsx
    interface Signal {
      aborted: boolean;
    }

    const useAbortablePromise = (
      asyncFn: (signal: Signal) => Promise<void>,
      dependencies: DependencyList,
    ) => {
      useEffect(() => {
        const signal = { aborted: false };
        asyncFn(signal);
        return () => {
          signal.aborted = true;
        };
      }, dependencies);
    };
```